Mathlib/Analysis/Fourier/FourierTransformDeriv.lean

```diff
@@ -152,17 +167,171 @@ theorem hasFDerivAt_fourierIntegral
+
+/-- Decomposing `fourierPowSMulRight L f v n` as a composition of continuous bilinear and
+multilinear maps, to deduce easily its continuity and differentiability properties. -/
+lemma fourierPowSMulRight_eq_comp {f : V → E} {v : V} {n : ℕ} :
+    fourierPowSMulRight L f v n = (- (2 * π * I))^n • smulRightL ℝ (fun (_ : Fin n) ↦ W) E
+      (compContinuousLinearMapLRight
+        (ContinuousMultilinearMap.mkPiAlgebra ℝ (Fin n) ℝ) (fun _ ↦ L v)) (f v) := rfl
+
+@[continuity, fun_prop]
+lemma _root_.Continuous.fourierPowSMulRight {f : V → E} (hf : Continuous f) (n : ℕ) :
+    Continuous (fun v ↦ fourierPowSMulRight L f v n) := by
+  simp_rw [fourierPowSMulRight_eq_comp]
+  apply Continuous.const_smul
+  apply (smulRightL ℝ (fun (_ : Fin n) ↦ W) E).continuous₂.comp₂ _ hf
+  exact Continuous.comp (map_continuous _) (continuous_pi (fun _i ↦ L.continuous))
+
+lemma _root_.ContDiff.fourierPowSMulRight {f : V → E} {k : ℕ∞} (hf : ContDiff ℝ k f) (n : ℕ) :
+    ContDiff ℝ k (fun v ↦ fourierPowSMulRight L f v n) := by
+  simp_rw [fourierPowSMulRight_eq_comp]
+  apply ContDiff.const_smul
+  apply (smulRightL ℝ (fun (_ : Fin n) ↦ W) E).isBoundedBilinearMap.contDiff.comp₂ _ hf
+  apply (ContinuousMultilinearMap.contDiff _).comp
+  exact contDiff_pi.2 (fun _ ↦ L.contDiff)
+
+lemma norm_fourierPowSMulRight_le (f : V → E) (v : V) (n : ℕ) :
+    ‖fourierPowSMulRight L f v n‖ ≤ (2 * π * ‖L‖) ^ n * ‖v‖ ^ n * ‖f v‖ := by
+  apply ContinuousMultilinearMap.opNorm_le_bound _ (by positivity) (fun m ↦ ?_)
+  calc
+  ‖fourierPowSMulRight L f v n m‖
+    = (2 * π) ^ n * ((∏ x : Fin n, |(L v) (m x)|) * ‖f v‖) := by
+      simp [_root_.abs_of_nonneg pi_nonneg, norm_smul]
+  _ ≤ (2 * π) ^ n * ((∏ x : Fin n, ‖L‖ * ‖v‖ * ‖m x‖) * ‖f v‖) := by
+      gcongr with i _hi
+      · exact fun i _hi ↦ abs_nonneg _
+      · exact L.le_opNorm₂ v (m i)
+  _ = (2 * π * ‖L‖) ^ n * ‖v‖ ^ n * ‖f v‖ * ∏ i : Fin n, ‖m i‖ := by
+      simp [Finset.prod_mul_distrib, mul_pow]; ring
+
+variable {L}
+
+variable [SecondCountableTopology V] [MeasurableSpace V] [BorelSpace V] {μ : Measure V}
+
+lemma _root_.MeasureTheory.AEStronglyMeasurable.fourierPowSMulRight
+    {n : ℕ} (hf : AEStronglyMeasurable f μ) :
+    AEStronglyMeasurable (fun v ↦ fourierPowSMulRight L f v n) μ := by
+  simp_rw [fourierPowSMulRight_eq_comp]
+  apply AEStronglyMeasurable.const_smul'
+  apply (smulRightL ℝ (fun (_ : Fin n) ↦ W) E).continuous₂.comp_aestronglyMeasurable₂ _ hf
+  apply Continuous.aestronglyMeasurable
+  exact Continuous.comp (map_continuous _) (continuous_pi (fun _i ↦ L.continuous))
+
+lemma integrable_fourierPowSMulRight {n : ℕ} (hf : Integrable (fun v ↦ ‖v‖ ^ n * ‖f v‖) μ)
+    (h'f : AEStronglyMeasurable f μ) : Integrable (fun v ↦ fourierPowSMulRight L f v n) μ := by
+  refine (hf.const_mul ((2 * π * ‖L‖) ^ n)).mono' h'f.fourierPowSMulRight ?_
+  filter_upwards with v
+  exact (norm_fourierPowSMulRight_le L f v n).trans (le_of_eq (by ring))
+
+variable (L)
+
+set_option maxHeartbeats 400000 in
```

Comment:
Do you know what takes so long?

Reply:
In fact it's not needed for the current version of the proof, so I've removed it. It was needed when I was building the proof, maybe because I hadn't squizzed some simps. My guess is that it's due to typeclass inference, because we're dealing with complicated objects (spaces of continuous linear maps from spaces of continuous multilinear maps to other spaces, for instance).